### ISSUE TITLE
[BUG] k3s clusters nodes are on unavailable state post rancher upgrade

### ISSUE THREAD

**Issue by anupama2501** [rancher/rancher#46061]
**Rancher Server Setup**
- Rancher version: v2.8.5 upgraded to v2.9-head `v2.9-1c2b81`
- Installation option (Docker install/Helm Chart): Docker Install

**Information about the Cluster**
- Kubernetes version: `v1.28.10+k3s1`
- Cluster Type (Local/Downstream): Downstream node driver k3s 1 node all roles, cloud provider: Linode and AWS

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom)
  - If custom, define the set of permissions: Admin

**Describe the bug**
Upon upgrading rancher from 2.8.5 to 2.9-head, notice the downstream k3s clusters are in active state but the nodes are in unavailable state.

**To Reproduce**
1. Create a rancher server on 2.8.5
2. Create a downstream k3s node driver cluster with 1 node all roles.
3. Upgrade rancher server to v2.9-head.

**Result**
Downstream cluster's node is in unavailable state.

**Expected Result**
Expected the node to come up active.

**Additional context**
Noticed the following logs in `capi-controller-manager` deployment:
```
E0708 20:57:51.095229       1 machineset_controller.go:892] "Unable to retrieve Node status" 
err="failed to create cluster accessor: error creating http client and mapper for remote cluster \"fleet-default/anuk3sotherrancher\": error creating client for remote cluster \"fleet-default/anuk3sotherrancher\": 
error getting rest mapping: failed to get API group resources: unable to retrieve the complete list of server APIs: v1: Get \"https://10.43.96.35/k8s/clusters/c-m-zw77pxst/api/v1?timeout=10s\": 
net/http: request canceled (Client.Timeout exceeded while awaiting headers)" controller="machineset" 
controllerGroup="cluster.x-k8s.io" controllerKind="MachineSet" MachineSet="fleet-default/anuk3sotherrancher-all-5b47f68bc9xddb6d" namespace="fleet-default" name="anuk3sotherrancher-all-5b47f68bc9xddb6d" reconcileID="3ea9fc61-187c-4bd9-bba2-437f4df3a295" 
MachineDeployment="fleet-default/anuk3sotherrancher-all" Cluster="fleet-default/anuk3sotherrancher" Machine="fleet-default/anuk3sotherrancher-all-5b47f68bc9xddb6d-6cb4m" node=""



E0708 20:57:15.482620       1 controller.go:329] "Reconciler error" err="failed to add &Node{ObjectMeta:{      0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [] [] []},Spec:NodeSpec{PodCIDR:,DoNotUseExternalID:,
ProviderID:,Unschedulable:false,Taints:[]Taint{},ConfigSource:nil,PodCIDRs:[],},Status:NodeStatus{Capacity:ResourceList{},Allocatable:ResourceList{},Phase:,Conditions:
[]NodeCondition{},Addresses:[]NodeAddress{},DaemonEndpoints:NodeDaemonEndpoints{KubeletEndpoint:DaemonEndpoint{Port:0,},},
NodeInfo:NodeSystemInfo{MachineID:,SystemUUID:,BootID:,KernelVersion:,OSImage:,ContainerRuntimeVersion:,KubeletVersion:,KubeProxyVersion:,OperatingSystem:,Architecture:,},Images:[]ContainerImage{},VolumesInUse:[],VolumesAttached:
[]AttachedVolume{},Config:nil,},} watch on cluster fleet-default/anuk3sotherrancher: 
failed to create cluster accessor: 
error creating http client and mapper for remote cluster \"fleet-default/anuk3sotherrancher\": error creating client for remote cluster \"fleet-default/anuk3sotherrancher\": error getting rest mapping: failed to get API group resources: unable to retrieve the complete list of server APIs: 
v1: Get \"https://10.43.96.35/k8s/clusters/c-m-zw77pxst/api/v1?timeout=10s\": net/http: request canceled (Client.Timeout exceeded while awaiting headers)" controller="machine" controllerGroup="cluster.x-k8s.io" 
controllerKind="Machine" Machine="fleet-default/anuk3sotherrancher-all-5b47f68bc9xddb6d-6cb4m" namespace="fleet-default" name="anuk3sotherrancher-all-5b47f68bc9xddb6d-6cb4m" reconcileID="2dfaee5a-62d4-42fe-ab33-9c3058fb1f7e"

```

### Note:

- Issue is not noticed on rancher HA using helm installs.
- Even on docker installs, issues is noticed only for k3s clusters and not rke2 or rke1 clusters.
- Issue is consistently happening on docker installs 2/2 times.

**Comment by snasovich** [comment]
@anupama2501 , there is a considerable chance https://github.com/rancher/rancher/pull/46086 done for https://github.com/rancher/rancher/issues/46062 may have fixed this. Please retest and let us know.

**Comment by markusewalker** [comment]
Validated that this is addressed on `v2.9-head` https://github.com/rancher/rancher/commit/3a6de11e4de9a4edaaf971ee5b91c6cc6f141200.

**ENVIRONMENT DETAILS**
- Rancher install: Docker
- Rancher version:
     - Fresh install: `v2.8.5`
     - Upgrade: `v2.9-head` https://github.com/rancher/rancher/commit/3a6de11e4de9a4edaaf971ee5b91c6cc6f141200

**TEST RESULT**
|#|Scenario|Result
|---|---|---|
|1|Upgrade Rancher 2.8 -> 2.9-head with downstream K3s node driver clusters|:white_check_mark:|
|2|Upgrade downstream K3s node driver clusters on upgraded Rancher|:white_check_mark:|
|3|Provision downstream K3s node driver cluster on upgraded Rancher|:white_check_mark:|

**VALIDATION STEPS**

**Scenario 1**
1. Setup Rancher v2.8.
2. Provisioned the following K3s v1.27 clusters:
     - 1 node all roles
     - etcd/cp node + 1 worker node
     - 1 etcd, 1 cp, 1 worker
3. Upgraded Rancher v2.8 -> v2.9-head.
4. Validated all clusters and nodes are `Active`.

______

**Scenario 2**
1. Following scenario 1, upgraded the K3s v1.27 clusters to `v1.30.2+k3s2`.
2. Validated that the clusters and nodes upgraded and are `Active`.

_______

**Scenario 3**
1. Following scenario 1, provisioned a K3s v1.30.2+k3s2 cluster (1 node all roles).
2. Validated the clsuter and nodes came up as `Active`.
